com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ClientReleaser.java

```diff
@@ -15,19 +14,20 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ClientReleaser.class);
 
-    private final FlowableEngineEvent event;
     private final CloudControllerClientProvider clientProvider;
 
-    public ClientReleaser(FlowableEngineEvent event, CloudControllerClientProvider clientProvider) {
-        this.event = event;
+    public ClientReleaser(CloudControllerClientProvider clientProvider) {
         this.clientProvider = clientProvider;
     }
 
-    public void releaseClient() {
+    public void releaseClientFor(String processInstanceId) {
         HistoryService historyService = Context.getProcessEngineConfiguration()
             .getHistoryService();
-        String processInstanceId = event.getProcessInstanceId();
 
+        releaseClientFor(historyService, processInstanceId);
+    }
+
+    public void releaseClientFor(HistoryService historyService, String processInstanceId) {

```

Comment:
Do we really need this to be public? Yes, it's good for tests, but it messes up the interface. Usually, we overload methods when it makes sense for users to customize one of the parameters. Here, however, it shouldn't make a difference whether the user supplies the `HistoryService` or uses the default one. They should be the exact same object.